### PR TITLE
Support sound null safety flag

### PR DESCRIPTION
**Pull request by grouma**
Add support for `sound-null-safety` flag with the following behavior:

If provided, `package:build_runner` will be run with sound null safety support. If negated, `package:build_runner` will be run without sound null safety support. If  not provided, the default `package:build_runner` behavior is used.

Closes https://github.com/dart-lang/webdev/issues/1217

### CODE REVIEW

File: webdev/README.md [hunk 1]
```diff
@@ -105,6 +105,14 @@ Common:
                                            (defaults to on)
 -e, --[no-]enable-expression-evaluation    Enable expression evaluation features
                                            in the debugger.
+    --[no-]sound-null-safety               If provided, `package:build_runner`
```
Comment (jakemac53) [inline webdev/README.md:108]: nit: it is really `package:build_web_compilers` which is being configured here

**Review by jakemac53 — COMMENTED** [review]
We should have an integration test here probably as well - see for instance this test https://github.com/dart-lang/build/blob/master/_test_null_safety/test/disable_sound_null_safety_test.dart. That should pass with `--no-sound-null-safety` and fail with `--sound-null-safety`.

**Comment by grouma** [comment]
> We should have an integration test here probably as well - see for instance this test https://github.com/dart-lang/build/blob/master/_test_null_safety/test/disable_sound_null_safety_test.dart. That should pass with --no-sound-null-safety and fail with --sound-null-safety.

Hmm. We don't really have that kind of infrastructure here. I can at the very least ensure that the right argument is passed to `package:build_runner`?

File: webdev/lib/src/command/configuration.dart [hunk 2]
```diff
@@ -243,6 +248,10 @@ class Configuration {
 
   bool get verbose => _verbose ?? false;
 
+  // Null indicates that the default `package:build_runner` behavior should be
+  // used.
```
Comment (jakemac53) [inline webdev/lib/src/command/configuration.dart:252]:
```suggestion
  // Null indicates that the default `package:build_web_compilers`
  // behavior should be used.
```

**Review by jakemac53 — APPROVED** [review]

**Pull request merged**